scripts/sexbound/lib/sexbound.lua

```diff
@@ -336,23 +338,48 @@ function Sexbound:addActor(actorConfig, store)
     --actor:getApparel():sync()
 
     --actor:initPlugins()
-    
-    if actor:getEntityType() == "player" then self._containsPlayer = true end
-    
+
     self._positions:filterPositions(self._actors)
     
     self._UI:refresh()
-    actor:openUI()
+    
+    if actor:getEntityType() == "player" then 
+        self._containsPlayer = true 
+        -- Open UI if not defeated or is allowed to while defeated.
+        if not actor:getStatus():hasStatus("sexbound_defeated") then
+            actor:openUI()
+            self._playerControl = true
```

Comment:
It appears you are missing the part where you actually set `self._containsDefeated`, which is used a couple lines further down.

Reply:
This gets set later lines 380-382. The checks on 358 and 372 are for when an actor joins a node that already contains a defeated actor.

Reply:
So, you are doing it this way to lock the node up once a second actor joins, assuming the defeated one is the first?

You also mentioned in the potential issues that this potentially disallows 3 actors and causes PVP problems with the defeated player interacting with their own node in time. There is a better solution to this problem. I don't know if there is a way to, via status effect, turn of a players ability to interact as a whole, but even if not you'd just need to edit the "onInteraction" function on the main node itself. The node upon creation gets the entity ID of the OG "defeated" target and, in the "onInteraction" function, if the interacting entity is that ID just null their lounge request. The "onInteraction" method is supposed to return `nil` or the name of an interaction type - the entity only lounges if the response is nil (and it's a loungable) or specifically the lounge operation. You can look up in the StarGlue repo what the allowed types are, or I can check it in a bit, cause I have done something similar for the Neki race mode whereas non-Nekis cannot lounge in a specific object.

Reply:
I did look into nullifying specific lounge requests using "onInteraction" but had no luck with my attempts. I think I had it returning nil expecting it to just skip the lounge handling but that makes sense. Am I right that what you're saying is that I could just have it do the entity ID check and have it return that exact message that you wrote?

Reply:
Yep. You just need to somehow transmit the source entity's ID on transformation, so the node inherently knows...
...actually, no you don't. You just block requests of the entity IDs of every actor that is already in there. The data is all there once the "lounging" is successful. Though that would still leave room for accidents with conveniently interacting the tick the node is spawned, so maybe still transmit the source entity's ID in the transformation itself, and lock that one out.

Returning the above will nullify the lounge request, because the interaction actually tries to trigger a message handler that should not exist, and hence does nothing.

Reply:
Seems clear to me! I'll work on that change.